ch_pipeline/core/io.py

```diff
@@ -473,3 +472,52 @@ def next(self, files):
         )
 
         return sorted(list(new_files))
+
+class SaveGaltAutoCorrelation(task.SingleTask):
+    """Extract the autocorrelations of the Galt telescope from a holography acquisition."""
+
+    # YY, YX, XX
+    _galt_prods = [2450, 2746, 3568]
+
+    def process(self, data):
+        """Extract the Galt autocorrelations and write them to disk.
+        Parameters
+        ----------
+        data: TimeStream
+            A TimeStream container holding a raw holography acquisition.
+        Returns
+        -------
+        autocorrelation: containers.GaltAutocorrelation
+            A GaltAutocorrelation container holding the extracted Galt autos
+            as a function of frequency, polarization product, and time.
+        """
+        from .containers import GaltAutocorrelation
+
+        # Redistribute over freq
+        data.redistribute("freq")
+
+        # Dereference beam and weight datasets
+        beam = data.vis[:].view(np.ndarray)
```

Comment:
the preferred way to this now is to use `local_array`
```suggestion
        beam = data.vis[:].local_array
```